packages/admin/cms-admin/src/generator/future/generateGrid.ts

```diff
@@ -827,3 +853,38 @@ export function generateGrid(
         gqlDocuments,
     };
 }
+
+const getDataGridComponentsProps = (forwardToolbarAction: boolean | undefined, excelExport: boolean | undefined) => {
+    if (!forwardToolbarAction && !excelExport) {
+        return "";
+    }
+
+    const values = `{
+        ${forwardToolbarAction ? `toolbarAction,` : ""}
+        ${excelExport ? `exportApi,` : ""}
+    }`.replace(/\n/g, "");
+
+    return `componentsProps={{
+        toolbar: ${values}
+    }}`;
+};
+
+const generateGridExportApi = (excelExport: boolean | undefined, gqlTypePlural: string, gridQuery: string) => {
+    if (!excelExport) {
+        return "";
+    }
+
+    return `const exportApi = useDataGridExcelExport<GQL${gqlTypePlural}GridQuery["${gridQuery}"]["nodes"][0], GQL${gqlTypePlural}GridQuery, GQL${gqlTypePlural}GridQueryVariables>({
+        columns,
+        // @ts-expect-error TODO: Figure out why this works in the handmade demo and fix it
```

Comment:
When do you intend to fix this?

Reply:
Hopefully some time today or tomorrow, I have a reminder for this, so I won't forget 😄

Reply:
Done in 87ad9052b694738126643b06e2938bcc2f27ecce